src/category_theory/lifting_properties.lean

```diff
@@ -0,0 +1,208 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Scholbach
+-/
+import category_theory.category
+import category_theory.arrow
+import category_theory.functor
+import category_theory.limits.preserves.shapes.terminal
+
+/-!
+# Lifting properties
+
+This file defines the lifting property of two arrows in a category and show basic properties of 
+this notion. 
+We also construct the subcategory consisting of those morphisms which have the right lifting 
+property with respect to arrows in a given diagram.
+
+## Main results
+- `has_lifting_property`: the definition of the lifting property
+- `iso_has_right_lifting_property`: any isomorphism satisfies the right lifting property (rlp)
+- `id_has_right_lifting_property`: any identity has the rlp
+- `right_lifting_property_initial_iff`: spells out the rlp against a map whose source is an 
+initial object
+- `right_lifting_subcat`: given a functor F : D ⥤ arrow C, we construct the subcategory of those 
+morphisms p in C that satisfy the rlp w.r.t. F(i), for any object i in D.
+
+## Tags
+lifting property
+-/
+
+namespace category_theory
+
+universes v u
+variables {C : Type u} [category.{v} C]
+variables {D : Type u} [category.{v} D]
+
+variables {X Y Z : C}
+
+/- The left lifting property of a morphism i vs. a morphism p. -/
+def has_lifting_property (i p : arrow C) : Prop := ∀ (sq : i ⟶ p), arrow.has_lift sq
+
+/- Any isomorphism has the right lifting property with respect to any map. 
+A    → X 
+↓i    ↓p≅
+B    → Y
+-/
+lemma iso_has_right_lifting_property (i : arrow C) (p : X ≅ Y) : 
+has_lifting_property i (arrow.mk p.hom) :=
+begin
+  intro sq,
+  fconstructor,
+  fconstructor,
+  fconstructor,
+  { use sq.right ≫ p.inv }, -- the lift is obtained by p⁻¹ ∘ (B → Y)
+  { rw [←category.assoc, (arrow.w sq).symm],
+    simp, },
+  { simp, }
+end
+
+/- Any identity has the right lifting property with respect to any map. -/
+lemma id_has_right_lifting_property (i : arrow C) : has_lifting_property i (arrow.mk (𝟙 X)) :=
+begin
+  haveI := is_iso.id X,
+  exact iso_has_right_lifting_property i (category_theory.as_iso (𝟙 X)),
+end
```

Comment:
```suggestion
iso_has_right_lifting_property i (category_theory.as_iso (𝟙 X))
```
(I could tell this would work because `is_iso.id` is already an instance, so `haveI` shouldn't be needed.

Reply:
Thanks. TwoFX already suggested a similar, slightly different solution here, which I adopted.